honey-token-tools/disseminate_honeytokens.py

```diff
@@ -0,0 +1,217 @@
+    CredentialsValidationError,
+    HoneyTokenCreationError,
+    PullRequestCreationError,
+    PullRequestInfo,
+    RepositoryAccessError,
+    RepositoryInfo,
+    get_branch_name_from_commit_message,
+    get_saas_url_for_vcs,
+)
+from gg_client import GGClient
+
+from utils import get_client_for_vcs
+
+
+def parse_vcs_url(vcs_url: str | None, vcs: str) -> str | None:
+    if not vcs_url:
+        vcs_url = os.getenv("VCS_URL")
+        if not vcs_url:
+            vcs_url = get_saas_url_for_vcs(vcs)
+    return vcs_url.strip("/")
+
+
+def parse_gitguardian_url(gitguardian_url: str | None) -> str | None:
+    if not gitguardian_url:
+        gitguardian_url = os.getenv("GITGUARDIAN_URL")
+        if not gitguardian_url:
+            gitguardian_url = GITGUARDIAN_SAAS
+    return gitguardian_url.strip("/")
+
+
+def validate_parameters(args: argparse.Namespace, parser: argparse.ArgumentParser):
+    if args.repo_names and args.repo_ids:
+        parser.error("Cannot use --repo-names and --repo-ids together.")
+
+    if not args.repo_names and not args.repo_ids:
+        parser.error("Must provide either --repo-names or --repo-ids.")
+
+    if not args.repo_names and args.vcs == "github":
+        parser.error("Should include --repo-names for GitHub.")
+
+    if not args.repo_ids and args.vcs == "gitlab":
+        parser.error("Should include --repo-ids for Gitlab.")
+
+    repos = args.repo_names if args.repo_names else args.repo_ids
+
+    vcs_url = parse_vcs_url(args.vcs_url, args.vcs)
+    vcs_token = os.getenv("VCS_TOKEN")
+
+    if not vcs_token:
+        parser.error(
+            "VCS token is required. Configure it via environment variable VCS_TOKEN."
+        )
+
+    gitguardian_url = parse_gitguardian_url(args.gitguardian_url)
+    gitguardian_token = os.getenv("GITGUARDIAN_TOKEN")
+
+    if not gitguardian_token:
+        parser.error(
+            "GitGuardian access token is required. Configure it via environment variable GITGUARDIAN_TOKEN."
+        )
+
+    return vcs_url, vcs_token, gitguardian_url, gitguardian_token, repos
+
+
+def main():
+    try:
+        import requests
+    except ImportError:
+        print("The 'requests' module should be installed to use this script.")
+        return
+
+    parser = argparse.ArgumentParser(
+        description="Script to disseminate honeytokens in your repositories via pull requests."
+    )
+
+    parser.add_argument(
+        "--vcs",
+        choices=["github", "gitlab"],
+        required=True,
+        help="Version control system",
+    )
+    parser.add_argument(
+        "--vcs_url",
+        help="VCS instance URL. If omitted, the default VCS URL will be used. "
+        "Can be also configured via environment variable VCS_URL.",
+    )
+    parser.add_argument(
+        "--gitguardian_url",
+        help="GitGuardian instance URL. If omitted, the default VCS URL "
+        "will be used. Can be also configured via environment variable GITGUARDIAN_URL.",
```

Comment:
```suggestion
        "--gitguardian_url",
        help="GitGuardian instance URL. If omitted, the default GitGuardian URL "
        "will be used. Can be also configured via environment variable GITGUARDIAN_URL.",
```
(Should we actually define what GitGuardian default URL is/means?)

Reply:
I agree, an example would be nice :) for instance, you could say, 'if omitted, https://dashboard.gitguardian.com will be used'